src/layers/rotary_embedding.h

```diff
@@ -35,16 +37,23 @@ class LlamaRotaryEmbedding {
 public:
     LlamaRotaryEmbedding(const int dim, const int max_position_embeddings = 2048, const float base = 10000);
 
-    ~LlamaRotaryEmbedding() {}
+    ~LlamaRotaryEmbedding();
 
-    void forward(float *query, float *key, int qStride, int kStride, const int *qkShape, const int *positionIds);
+    void forward(float *query, float *key, int qStride, int kStride, const int *qkShape, const int *positionIds,
```

Comment:
1. qk_shape already have seq_len, maybe we can add max_seq_length in the qk_shape to avoid upgrading every rotaryEmbedding implementation.
2. The NTK function is only for QWEN, maybe we can create a new implementation for QWEN instead of modifying the original LLAMA implementation..